File: projects/order-creation-queries/app/src/main/java/com/robinfood/app/usecases/getorderhistory/IGetOrderHistoryUseCase.java
package com.robinfood.app.usecases.getorderhistory;

import com.robinfood.core.dtos.EntityDTO;
import com.robinfood.core.dtos.orderhistory.request.OrderHistoryRequestDTO;
import com.robinfood.core.dtos.orderhistory.response.OrderHistoryItemDTO;
import com.robinfood.core.enums.Result;

/**
 * Use case that gets the order history
 */
public interface IGetOrderHistoryUseCase {

    /**
     * Retrieves the order history based on the following query params
     * @param orderHistoryRequestDTO information for the different filters
     * @return the order history containing the orders detailed info
     */
    Result<EntityDTO<OrderHistoryItemDTO>> invoke(OrderHistoryRequestDTO orderHistoryRequestDTO);
}

File: projects/posv2/app/src/main/kotlin/com/robinfood/app/usecases/getorderhistory/IGetOrderHistoryUseCase.kt
package com.robinfood.app.usecases.getorderhistory

import com.robinfood.core.dtos.HistoryDTO
import com.robinfood.core.dtos.orderhistory.OrderHistoryRequestDTO
import com.robinfood.core.enums.Result

/**
 * Use case that retrieves the order history
 */
interface IGetOrderHistoryUseCase {

    suspend fun invoke(
        token: String,
        orderHistoryRequestDTO: OrderHistoryRequestDTO
    ): Result<HistoryDTO>
}